testing/sdk_marathon.py

```diff
@@ -212,52 +200,58 @@ def _update():
             log_args=False,
             raise_on_error=False,
         )
-        return _handle_marathon_deployment_response(response)
+        return MarathonDeploymentResponse(response)
 
     result = _update()
-    result.raise_on_error()
```

Comment:
There is still a subtle difference here. We don't check for error messages.

Reply:
This should be now handled by the `MarathonDeploymentResponse` class during initialization. It always serializes the JSON first and logs it. Thus the error message is shown. Afterwards, it raises the HTTP error.

Reply:
Yes, but the original method raised an error if the error string in the message body was non-empty, or am I missing something?

Reply:
I see this wasn't clear from my side. Marathon only adds a message string in the response if the status code is not `2xx`. In this case the error is thrown immediately during initialization now. Therefore I don't think it is necessary anymore to explicitly check for the message since it should have the same meaning as the HTTP error code. The log statement showing the response JSON should in an error case now include the message string to help debugging.

Reply:
Ok. That's fine then! Thanks for the clarification.

Reply:
(do we log the error message as it would have been to add enough context?)